keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -138,16 +150,84 @@ void emptyTagFiltersReturnAllSecrets() {
   void nonExistentTagFiltersReturnsEmpty() {
     final Collection<SimpleEntry<String, String>> secretEntries =
         awsSecretsManagerExplicit.mapSecrets(
-            List.of("nonexistent-tag-key"), List.of("nonexistent-tag-value"), SimpleEntry::new);
+            Collections.emptyList(),
+            List.of("nonexistent-tag-key"),
+            List.of("nonexistent-tag-value"),
+            SimpleEntry::new);
+
+    assertThat(secretEntries).isEmpty();
+  }
+
+  @Test
+  void emptyPrefixFiltersReturnAllSecrets() {
+    final Collection<SimpleEntry<String, String>> secretEntries =
+        awsSecretsManagerExplicit.mapSecrets(
+            Collections.emptyList(),
+            Collections.emptyList(),
+            Collections.emptyList(),
+            SimpleEntry::new);
+
+    assertThat(secretEntries.stream().map(SimpleEntry::getKey))
+        .contains(
+            SECRET_NAME1_KEY1_VALA,
+            SECRET_NAME2_KEY1_VALB,
+            SECRET_NAME3_KEY2_VALC,
+            SECRET_NAME4_KEY2_VALB);
+    assertThat(secretEntries.stream().map(SimpleEntry::getValue))
+        .contains(SECRET_VALUE1, SECRET_VALUE2, SECRET_VALUE3, SECRET_VALUE4);
+  }
+
+  @Test
+  void nonExistentPrefixFilterReturnsEmpty() {
+    final Collection<SimpleEntry<String, String>> secretEntries =
+        awsSecretsManagerExplicit.mapSecrets(
+            List.of("nonexistent-secret-prefix"),
```

Comment:
If we combine nonexistent-secret-prefix with a tag that exists, do we still get empty? i.e. are the filters combined with an 'AND'?

Reply:
It is an AND. I tested it but didn't add a test, I'll add it now.

Reply:
done